src/route.rs

```diff
@@ -138,6 +140,33 @@ impl<'a, State: 'static> Route<'a, State> {
         Ok(())
     }
 
+    /// Serve a index file statically.
+    ///
+    /// The file will be streamed from disk, and a mime type will be determined
+    /// based on magic bytes.
+    ///
+    /// # Examples
+    ///
+    /// Serve the index file at `./public/images/index.json` from
+    /// `localhost:8080/images/`.
+    ///
+    /// ```no_run
+    /// #[async_std::main]
+    /// async fn main() -> Result<(), std::io::Error> {
+    ///     let mut app = tide::new();
+    ///     app.at("/public/images").serve_index("images/index.json")?;
+    ///     app.listen("127.0.0.1:8080").await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn serve_index(&mut self, dir: impl AsRef<Path>) -> io::Result<()> {
```

Comment:
Is `dir` intended to be a single file? It's confusing that it's named `dir` and we use ServeDir to resolve a single file instead of just `Ok(Body::from_file(index_path).await?)`